modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java

```diff
@@ -0,0 +1,139 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.testcontainers.elasticsearch;
+
+
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.client.RestClient;
+import org.junit.After;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertThrows;
+
+public class ElasticsearchContainerTest {
+
+    private ElasticsearchContainer container = null;
+    private RestClient client = null;
+
+    @After
+    public void stopRestClient() throws IOException {
+        if (client != null) {
+            client.close();
+            client = null;
+        }
+    }
+
+    @After
+    public void stopContainer() {
+        if (container != null) {
+            container.stop();
+        }
+    }
+
+    @Test
+    public void elasticsearchNoVersionTest() throws IOException {
+        container = new ElasticsearchContainer();
+        container.start();
+        Response response = getClient(container).performRequest("GET", "/");
+        assertThat(response.getStatusLine().getStatusCode(), is(200));
+
+        // The default image is running with the features under Elastic License
+        response = getClient(container).performRequest("GET", "/_xpack/");
+        assertThat(response.getStatusLine().getStatusCode(), is(200));
+        // For now we test that we have the monitoring feature available
+        assertThat(EntityUtils.toString(response.getEntity()), containsString("monitoring"));
+    }
+
+    @Test
+    public void elasticsearchDefaultTest() throws IOException {
+        container = new ElasticsearchContainer();
+        container.withVersion(ElasticsearchContainer.ELASTICSEARCH_DEFAULT_VERSION);
+        container.start();
+        Response response = getClient(container).performRequest("GET", "/");
+        assertThat(response.getStatusLine().getStatusCode(), is(200));
+    }
+
+    @Test
+    public void elasticsearchVersion() throws IOException {
```

Comment:
Isn't this effectively the same than `elasticsearchDefaultTest`, since it will fallback to the default version?

Reply:
No. The version we are testing in that case is `${elasticsearchOldVersion}` which is evaluated as `5.6.10`. (Unless I did something wrong :) )

Reply:
This is more obvious with the static String I'm now using.
That said, I removed indeed some unneeded tests as I'm not using anymore property files.